toolkit/scripts/check_new_kernel_configs.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+from kernel_sources_analysis import get_data_from_config, extract_kernel_dir_name, extract_config_arch
+
+# Regex for finding config options
+# Matches words that start with +/-CONFIG_ or +/-# CONFIG_
```

Comment:
I'd suggest to also add a few lines of examples with lines that will match and the ones that won't. Usually that helps a lot with understanding regular expressions. Since we're using backward lookups here, I'd include also the final string that's going to end up in the match. Something along these lines
```
Examples:
"+CONFIG_ABC=y" -> "CONFIG_ABC"
"-# CONFIG_XYZ is not..." -> "CONFIG_XYZ"
```